### PR TITLE
feat: [M3-7141] Add codeowner file to monorepo root

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
Small PR to add a CODEOWNER file needed for the auto assignment of PRs

There's not much to test, once merged it should trigger the rules configured in the repo's settings.

No need for a changeset here either.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default code owners
+* @frontend
```
Comment (abailly-akamai) [inline CODEOWNERS:2]: for context, @frontend is our github team and the wildcard is for the entire repo to be owned by that team. 

We can add granularity as needed if we want to assign parts of the code base to sub groups or individuals.

**Note**: this only affects assignment of Pull Requests. 

> **Code Review and Approval**: Code owners or teams specified in the CODEOWNERS file are responsible for reviewing and approving changes within their designated areas, but their approval alone does not automatically merge a pull request.
> 
> **Merging Pull Requests**: Anyone with write access to the repository, regardless of whether they are part of the code owners or teams, can merge pull requests. This includes repository administrators, users with write permissions, and even those who are not specified in the CODEOWNERS file.

Comment (bnussman-akamai) [inline CODEOWNERS:2]: Maybe it should be `@linode/frontend`? I don't know anything about this feature so I'm not 100% sure

Screenshot 2023-09-15 at 1 36 23 PM

Comment (abailly-akamai) [reply CODEOWNERS:2]: > You do not need to include the organization name (linode) as part of the team reference in the CODEOWNERS file. GitHub identifies teams globally within your organization, so specifying @frontend is sufficient to reference the "frontend" team within the "linode" organization.

> The @ symbol followed by the team name is all you need to reference the team in the CODEOWNERS file, regardless of the organization context.

Comment (abailly-akamai) [reply CODEOWNERS:2]: am looking into this and I keep finding different answers

Comment (abailly-akamai) [reply CODEOWNERS:2]: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

> Teams can be specified as code owners as well. Teams should
# be identified in the format @org/team-name. Teams must have
# explicit write access to the repository. In this example,
# the octocats team in the octo-org organization owns all .txt files.
*.txt @octo-org/octocats

will change to @linode/frontend

Comment (bnussman-akamai) [reply CODEOWNERS:2]: So you’re saying Github's warning is wrong?

Comment (abailly-akamai) [reply CODEOWNERS]: am saying the opposite. what i found was outdated

**Review by bnussman-akamai — APPROVED** [review]

**Review by cpathipa — APPROVED** [review]

**Pull request merged**